scipy/optimize/slsqp.py

```diff
@@ -249,17 +254,50 @@ def _minimize_slsqp(func, x0, args=(), jac=None, bounds=None,
         iprint = 0
 
     # Transform x0 into an array.
-    x = asfarray(x0).flatten()
+    full_x0 = np.copy(asfarray(x0).flatten())
```

Comment:
A lot of this is repeated from above, and it may have to be done for `trust-constr` at some point. Can the code be shared?

Reply:
When the changes were made that caused the issue for SLSQP and LBFGSB they didn't affect `trust-constr`, the behaviour of `trust-constr` w.r.t equal bounds has been unchanged since its inception.
`trust-constr` is quite complicated, with a lot more user specifiable options and output. I think the approach may be able to be shared, but I'm not sure about code.
In this PR I'm not sure if I could find any more code in the changes to SLSQP and LBFGSB that could be factored out into shared functions.